platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/SyncUtils.java

```diff
@@ -478,22 +428,27 @@ private static boolean haveWeKnownAboutEventForALongTime(
     }
 
     /**
-     * Returns a predicate that determines if a {@link ShadowEvent}'s generation is non-ancient for the peer and greater
-     * than this node's minimum non-expired generation, and is not already known.
+     * Returns a predicate that determines if a {@link ShadowEvent}'s ancient indicator is non-ancient for the peer and
+     * greater than this node's minimum non-expired threshold, and is not already known.
      *
      * @param knownShadows     the {@link ShadowEvent}s that are already known and should therefore be rejected by the
      *                         predicate
-     * @param myGenerations    the generations of this node
-     * @param theirGenerations the generations of the peer node
+     * @param myEventWindow    the event window of this node
+     * @param theirEventWindow the event window of the peer node
+     * @param ancientMode      the current ancient mode
      * @return the predicate
      */
+    @NonNull
     public static Predicate<ShadowEvent> unknownNonAncient(
-            final Collection<ShadowEvent> knownShadows,
-            final GraphGenerations myGenerations,
-            final GraphGenerations theirGenerations) {
-        long minSearchGen =
-                Math.max(myGenerations.getMinRoundGeneration(), theirGenerations.getMinGenerationNonAncient());
-        return s -> s.getEvent().getGeneration() >= minSearchGen && !knownShadows.contains(s);
+            @NonNull final Collection<ShadowEvent> knownShadows,
+            @NonNull final NonAncientEventWindow myEventWindow,
+            @NonNull final NonAncientEventWindow theirEventWindow,
+            @NonNull final AncientMode ancientMode) {
+
+        final long minimumSearchThreshold =
```

Comment:
Mostly, this is just a continuance of the previous logic... I don't think the `max()` is actually required. This should always be evaluating to the peer's ancient threshold. But since this is a simple check, I think it's safer to leave it in place for the time being.

I've added the following comment.

```
        // When searching for events, we don't want to send any events that are known to be ancient to the peer.
        // We should never be syncing with a peer if their ancient threshold is less than our expired threshold 
        // (if this is the case, then the peer is "behind"), so in practice the minimumSearchThreshold will always 
        // be the same as the peer's ancient threshold. However, in an abundance of caution, we use the maximum of 
        // the two thresholds to ensure that we don't ever attempt to traverse over events that are expired to us, 
        // since those events may be unlinked and could cause race conditions if accessed.
```